src/lib.rs

```diff
@@ -480,12 +480,17 @@ pub struct Guard<'l, T: RefCnt> {
 }
 
 impl<'a, T: RefCnt> Guard<'a, T> {
-    /// Turns the given guard into an unprotected, independent one.
-    ///
-    /// It'll not hold any locks, will have no debts and will contain full-featured value inside
-    /// that even can outlive the ArcSwap it originated from.
+    fn new(ptr: *const T::Base, protection: Protection<'a>) -> Guard<'a, T> {
+        Guard {
+            inner: ManuallyDrop::new(unsafe { T::from_ptr(ptr) }),
+            protection,
+        }
+    }
+
+    /// Drop any debt and release any lock held by the given guard and return a
+    /// full-featured value that even can outlive the ArcSwap it originated from.
     #[inline]
-    fn unprotected(mut lease: Self) -> Guard<'static, T> {
+    fn inc_and_unprotect(mut lease: Self) -> T {
```

Comment:
This particular change feels a bit fishy. First, even the name suggests this does two unrelated things, which is usually a red flag. Furthermore, it now does exactly the same as `into_inner` and we don't really need two identical methods.

The `new` method/constructor seems fine, though.

Style nitpick: documentation comments, similar to git commits, uses the first line as a caption. That means it should be short and single-line 😇.

Reply:
That was actually on purpose, since the "inc" part is what allows to "unprotect". On the other hand it is imprecise because it does not "inc" if the guard is Unprotected. I kept it as a separate method from `into_inner` to allow marking it `#[inline]`; this benefits the `load` case, where protection is always Locked, but I will merge them and make `into_inner` inline.